data-access/src/routes/http/graphql/schema/types/community.resolvers.ts

```diff
@@ -15,44 +17,95 @@ const CommunityMutationResolver = async (getCommunity: Promise<CommunityDo>): Pr
   }
 };
 
+const checkPermission = (passport, community, permissionCheck) => {  
+  return passport.datastoreVisa.forCommunity(community as CommunityDo).determineIf(permissionCheck);  
+};  
+
 const community: Resolvers = {
   Community: {
-    roles: async (_rootObj: Community, _, { applicationServices }) => {
-      return (await applicationServices.roles.endUserRole.dataApi.getRoles()) as Role[];
+    roles: async (rootObj: Community, _, { applicationServices, verifiedUser }) => {
+      if (checkStaffPortalAccess(verifiedUser)) {
+        return (await applicationServices.roles.endUserRole.dataApi.getRolesByCommunityId(rootObj.id.toString())) as Role[];
+      } else if (checkAccountPortalAccess(verifiedUser) || checkSystemAccess(verifiedUser)) {
+        return (await applicationServices.roles.endUserRole.dataApi.getRoles()) as Role[];
+      }
+      return [];
```

Comment:
For queries and fields on graphql types, I think its simpler and easier on the UI to return null or empty arrays for fields that the user does not have access to. For mutations, we throw error messages to alert the UI that the mutation failed. But for queries, some fields we may want to be public and shown to all authenticated users regardless of their permissions, while some specific fields may be gatekept behind permissions. So instead of having the query throw an error and break the UI, it only returns the fields that the user is allowed to see to the UI